Source/WebCore/rendering/RenderLayerScrollableArea.cpp

```diff
@@ -1476,8 +1480,14 @@ void RenderLayerScrollableArea::paintScrollCorner(GraphicsContext& context, cons
 
     // We don't want to paint a corner if we have overlay scrollbars, since we need
     // to see what is behind it.
-    if (!hasOverlayScrollbars())
+    if (!hasOverlayScrollbars()) {
+#if PLATFORM(MAC)
+        // Keep this in sync with ScrollAnimatorMac's effectiveAppearanceForScrollerImp:.
+        LocalDefaultSystemAppearance localAppearance(useDarkAppearance());
+#endif
```

Comment:
If possible, we should try to avoid adding platform-specific code in a platform-agnostic file.

In this case, I think we can:
1. Move this logic in to `ScrollbarThemeMac::paintScrollCorner`. Name the `ScrollableArea` parameter, and read `area.useDarkAppearanceForScrollbars()`.
2. Remove similar logic from `LocalFrameView::paintScrollCorner`.

If that doesn't work, we can move ahead with this solution.

Reply:
Taking a look at that now can't see why it wouldn't work but will test it